src/d3d9/d3d9_swapchain.h

```diff
@@ -197,6 +198,17 @@ namespace dxvk {
 
     std::string GetApiName();
 
+    const Com<D3D9Surface, false>& GetFrontBuffer() const {
+      return m_backBuffers.back();
+    }
+
+    bool HasFrontBuffer() const {
+      // Tests show that SWAPEEFFECT_DISCARD + 1 backbuffer in windowed mode behaves identically to SWAPEFFECT_COPY
+      // For SWAPEFFECT_COPY we don't swap buffers but do another blit to the front buffer instead.
+      return m_presentParams.SwapEffect != D3DSWAPEFFECT_COPY
```

Comment:
Split into multiple if's so this is just easier to follow. This chain of logic is ough enough as it is.